File: 2023/challenge_02/go/main.go
package main

import (
	"fmt"
	"io"
	"net/http"
	"strings"
)

func main() {
	URL := "https://codember.dev/data/message_02.txt"

	var VALUE int = 0

	resp, err := http.Get(URL)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	var builder strings.Builder
	_, err = io.Copy(&builder, resp.Body)
	if err != nil {
		panic(err)
	}

	content := builder.String()

	for i := 0; i < len(content); i++ {
		item := string(content[i])

		if item == "#" {
			VALUE += 1
		} else if item == "@" {
			VALUE -= 1
		} else if item == "*" {
			VALUE *= VALUE
		} else if item == "&" {
			fmt.Print(VALUE)
		} else {
		}
	}
}

File: 2023/challenge_02/python/main.py
import requests

URL = 'https://codember.dev/data/message_02.txt'

req = requests.get(URL)

SECRET = req.text
VALUE = 0

for item in SECRET:
  if item == '#':
    VALUE = VALUE + 1
  elif item == '@':
    VALUE = VALUE - 1
  elif item == '*':
    VALUE = VALUE * VALUE
  elif item == '&':
    #print(f"current value: {VALUE}")
    print(VALUE, end='')
  else:
    pass
  
#print(f"\nfinal value {VALUE}")